### ISSUE TITLE
HDMI sound not working on fresh installation

### ISSUE THREAD

**Issue by BrozzSama** [jit06/retroarch_fbdev_c1#2]
HDMI sound seems to be not working on a fresh installation of GamOdroid on ODROID-C1. The ALSA driver just crashes, while SDL works but with choppy sound.

Attached you can find the log of retroarch when running a game using VBA and Alsa driver. I tried both alsa and alsathread and didn't seem to make a difference.

[retroarch__2021_11_02__22_14_39.log](https://github.com/jit06/retroarch_fbdev_c1/files/7574323/retroarch__2021_11_02__22_14_39.log)

**Comment by jit06** [comment]
Hello,

Do you confirm that you have installed the V3 HDMI image (GamOdroiD-V3.0-HDMI.img.gz) ?

**Comment by jit06** [comment]
ok I checked and I saw in DMESG that the audio codec cannot be registered. Probably due to a missing module (snd_hdmi or somehitng like that)

I need to check that in the kernel part. Sorry but I'm unsure when I will have time to dig into that

**Comment by jit06** [comment]
I tried a simple wav with aplay and it works and if you choose sdl2 as audio driver in retroarch it works also.
There is definitly somehting that is missing for alsa driver to works in retroarch, but I can't manage to get a more verbose debug log (the only thing I can see if "failed to initialize" :(

**Comment by BrozzSama** [comment]
> I tried a simple wav with aplay and it works and if you choose sdl2 as audio driver in retroarch it works also.
> There is definitly somehting that is missing for alsa driver to works in retroarch, but I can't manage to get a more verbose debug log (the only thing I can see if "failed to initialize" :(

Yes, now that you mention it I did test it with aplay and it seemed to be working on my installation. As well, I wonder what has changed in your build that breaks the audio in retroarch. I don’t have the odroid with me right now, I’ll do some more testing next weekend.
